sidecar/src/service/tracing/trace_flusher.rs

```diff
@@ -131,6 +158,12 @@ impl TraceFlusher {
             send_data_size: self.inner.lock().unwrap().traces.send_data_size as u32,
         }
     }
+    
+    pub fn collect_metrics(&self) -> TraceFlusherMetrics {
```

Comment:
I know it's technically not mutable, but I'm not sure the name `collect_metrics` conveys the actual mutability of this function on `TraceFlusherMetrics`? Also, I don't think it's actually collecting metrics at this point (the caller is doing the collecting)?

What about calling the function `remove()`, similar to the [remove function in HashMap](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.remove)? Or something like `fetch_and_clear`?